core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -193,13 +193,7 @@ abstract class RawModule extends BaseModule {
     _component
   }
 
-  private[chisel3] def secretConnection(left: Data, _right: Data)(implicit si: SourceInfo): Unit = {
-    val (right: Data, _) = chisel3.experimental.dataview
-      .reifyIdentityView(_right)
-      .getOrElse(
-        throwException(s"BoringUtils currently only support identity views, ${_right} has multiple targets.")
-      )
```

Comment:
It felt weird to me that this check was in RawModule, the reason being this is like the final function called by BoringUtils. IMO it's better to do this earlier. All calls to `secretConnection` in the the method where I moved the reification.